src/utils/resize-watcher.js

```diff
@@ -19,6 +19,7 @@ class ResizeWatcher extends FakeEventTarget {
    * @returns {void}
    */
   destroy(): void {
+    // todo sakal Oren I think it should also remove all listeners, you might have memory leaks
```

Comment:
disconnect does unobserv and clears watcher: https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver/disconnect
For iframe api we clear the `onresize` ourselves.
Which event listener is left?

Reply:
I might be wrong here, this class inherit from `FakeEventTarget` which handles its own queue of listeners. So while you unobserve the window you still hold reference to handler functions of components.

Reply:
`FakeEventTarget` is the base object, it mimics the native DOM event target.
It is used by an event manager to register and unregister from listeners.
It is the responsibility of the listener invocation to unregister, which is done in player.js.